File: problem-arena-frontend/src/model/PaginationDTO.ts
export interface PaginationDTO {
    first_id: number,
    last_id: number,
    direction: number,
    limit: number
}

export interface ParticipationPaginationDTO {
    first_uid: number,
    first_cid: number,
    last_uid: number,
    last_cid: number,
    direction: number,
    limit: number
}

export interface StatisticPagination {
    first_stat: number,
    first_id: number,
    last_stat: number,
    last_id: number,
    direction: number,
    limit: number
}

File: backend/src/model/dto/pagination_dto.rs
use serde::Deserialize;

#[derive(Deserialize)]
pub struct PaginationDTO {
    pub first_id: i32,
    pub last_id: i32,
    pub direction: i32,
    pub limit: i32
}

#[derive(Deserialize)]
pub struct ParticipationPaginationDTO {
    pub first_uid: i32,
    pub first_cid: i32,
    pub last_uid: i32,
    pub last_cid: i32,
    pub direction: i32,
    pub limit: i32
}